tools/isobuild/linker.js

```diff
@@ -37,242 +39,30 @@ var packageDot = function (name) {
-    var self = this;
-
-    // The assigned variables in the app aren't actually used for anything:
-    // we're using the global namespace, so there's no header where we declare
-    // all of the assigned variables as vars.  So there's no use wasting time
-    // running static analysis on app code.
-    if (self.useGlobalNamespace) {
-      return [];
-    }
-
-    // Find all global references in any files
-    var assignedVariables = [];
-    for (const file of self.files) {
-      assignedVariables = assignedVariables.concat(
-          await file.computeAssignedVariables());
-    }
-    assignedVariables = _.uniq(assignedVariables);
-
-    return assignedVariables;
-  }),
-
-  // Output is a list of objects with keys 'source', 'servePath', 'sourceMap',
-  // 'sourcePath'
-  getPrelinkedFiles: Profile("linker Module#getPrelinkedFiles", async function () {
-    var self = this;
-
-    const haveMeteorInstallOptions =
-      self.files.some(file => file.meteorInstallOptions);
-
-    // If we don't want to create a separate scope for this module,
-    // then our job is much simpler. And we can get away with
-    // preserving the line numbers.
-    if (self.useGlobalNamespace &&
-        ! haveMeteorInstallOptions) {
-      // Ignore lazy files unless we have a module system.
-      const eagerFiles = _.filter(self.files, file => ! file.lazy);
-
-      const ret = [];
-      for (const file of eagerFiles) {
-        const cacheKey = JSON.stringify([
-          file._inputHash,
-          file.bare,
-          file.servePath,
-        ]);
-
-        if (APP_PRELINK_CACHE.has(cacheKey)) {
-          ret.push(APP_PRELINK_CACHE.get(cacheKey));
-        }
-
-        const node = await file.getPrelinkedOutput({ preserveLineNumbers: true });
-        const results = await Profile.time(
-            "toStringWithSourceMap (app)", () => {
-              return node.toStringWithSourceMap({
-                file: file.servePath
-              }); // results has 'code' and 'map' attributes
-            }
-        );
-
-        let sourceMap = results.map.toJSON();
-        if (! sourceMap.mappings) {
-          sourceMap = null;
-        }
-
-        const prelinked = {
-          source: results.code,
-          sourcePath: file.sourcePath,
-          servePath: file.servePath,
-          sourceMap: sourceMap,
-          hash: file._inputHash,
-        };
-
-        APP_PRELINK_CACHE.set(cacheKey, prelinked);
-        ret.push(prelinked);
-      }
-
-      // TODO[fibers]: This is a temporary hack to make sure that we don't return the same
-        // file twice. I'm not sure why, but self.files sometimes contains the same file twice.
-        // these tool tests fail without this hack:
-        // - assets - unicode asset names are allowed
-        // - javascript hot code push
-        // and probably others
-      const uniqueHashes = [...new Set(ret.map((f) => f.hash))];
-      return uniqueHashes.map((h) => ret.find((f) => f.hash === h));
```

Comment:
The bug that caused this was introduced when the for ... of loop above was added when removing fibers - it didn't handle caching properly and would add duplicate files when there was a cached entry.

This code was also incompatible with TLA since it generated multiple files for the app. It did a simplified form of linking for apps that did not use modules, which is rare. This PR removes it in favor of linking them in the same way as packages that do not use modules.